compiler/rustc_resolve/src/lib.rs

```diff
@@ -2044,10 +2072,14 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
     }
 
     #[inline]
-    fn add_to_glob_map(&mut self, import: Import<'_>, ident: Ident) {
+    fn add_to_glob_map<'r>(
+        mut self: SmartResolver<'r, 'ra, 'tcx>,
```

Comment:
Also never called from import resolution.
Could you audit all of the converted functions and keep only those that are strictly necessary?

The algorithm from https://github.com/rust-lang/rust/pull/144912#issuecomment-3157496400 should work.
Put `res_mut` at the top in `resolve_import`, if it panics - move it one level lower, repeat.